avro_derive/src/attributes/mod.rs

```diff
@@ -0,0 +1,202 @@
+            || serde.variant_identifier
+            || serde.field_identifier
+        {
+            errors.push(syn::Error::new(
+                span,
+                "AvroSchema derive does not support changing the tagging Serde generates (`tag`, `content`, `untagged`, `variant_identifier`, `field_identifier`)",
+            ));
+        }
+        if serde.remote.is_some() {
+            errors.push(syn::Error::new(
+                span,
+                "AvroSchema derive does not support the Serde `remote` attribute",
+            ));
+        }
+        if serde.transparent {
+            errors.push(syn::Error::new(
+                span,
+                "AvroSchema derive does not support Serde transparent",
+            ))
+        }
+        if serde.rename_all.deserialize != serde.rename_all.serialize {
+            errors.push(syn::Error::new(
+                span,
+                "AvroSchema derive does not support different rename rules for serializing and deserializing (`rename_all(serialize = \"..\", deserialize = \"..\")`)"
+            ))
+        }
+
+        // Check for conflicts between Serde and Avro
+        if avro.rename_all != RenameRule::None && serde.rename_all.serialize != avro.rename_all {
+            errors.push(syn::Error::new(
+                span,
+                "#[avro(rename_all = \"..\")] must match #[serde(rename_all = \"..\")], it's also deprecated. Please use only `#[serde(rename_all = \"..\")]`",
+            ))
+        }
+
+        if !errors.is_empty() {
+            return Err(errors);
+        }
+
+        Ok(Self {
+            name: avro.name,
+            namespace: avro.namespace,
+            doc: avro.doc,
+            alias: avro.alias,
+            rename_all: serde.rename_all.serialize,
+        })
+    }
+}
+
+pub struct FieldOptions {
+    pub doc: Option<String>,
+    pub default: Option<String>,
+    pub alias: Vec<String>,
+    pub rename: Option<String>,
+    pub skip: bool,
+    pub flatten: bool,
+}
+
+impl FieldOptions {
+    pub fn new(attributes: &[Attribute], span: Span) -> Result<Self, Vec<syn::Error>> {
+        let avro = avro::FieldAttributes::from_attributes(attributes).map_err(darling_to_syn)?;
+        let serde = serde::FieldAttributes::from_attributes(attributes).map_err(darling_to_syn)?;
+
+        // Collect errors so user gets all feedback at once
+        let mut errors = Vec::new();
+
+        // Check for any Serde attributes that are hard errors
+        if serde.getter.is_some() {
+            errors.push(syn::Error::new(
+                span,
+                "AvroSchema derive does not support the Serde `getter` attribute",
+            ));
+        }
+
+        // Check for conflicts between Serde and Avro
+        if avro.skip && !serde.skip {
+            errors.push(syn::Error::new(
+                span,
+                "`#[avro(skip)]` requires `#[serde(skip)]`, it's also deprecated. Please use only `#[serde(skip)]`"
```

Comment:
Is there a way to report the usage of a deprecated `avro(attribute)` as a warning without depending on the value of its respective `serde` counter-part ?
I.e. pseudo code:
```
if avro.skip { report a warning like "#[avro(skip)] is deprecated. Use #[serde(skip)] instead" }

if avro.skip != serde.skip {report an error, without the ", it's also deprecated" text}
```

Currently the user will be notified that a deprecated avro attribute is used only if the avro value differs from the serde's one.

Reply:
Sounds good!
Let's do it!
Then once it is in `stable` and our MSRV is at least this stable version we could delete just this attribute to enable it for all.